File: packages/installer-server/src/lib/steps/verify.ts
import { execa } from 'execa';
import type { InstallerStep } from './index.js';

export const verifyStep: InstallerStep = {
  id: 'verify',
  label: 'Run `openclaw doctor`',
  shouldRun: () => true,
  async run({ bus }) {
    try {
      const { stdout } = await execa('openclaw', ['doctor'], { all: true });
      bus.log(verifyStep.id, 'info', stdout || 'doctor reported no issues.');
    } catch (err) {
      const message = err instanceof Error ? err.message : String(err);
      // Doctor exits non-zero on warnings too — surface it but don't fail
      // the entire pipeline; the user should see the report and decide.
      bus.log(verifyStep.id, 'warn', `doctor reported issues: ${message}`);
    }
  },
};

File: internal/installer/steps/verify.go
package steps

import (
	"context"
	"os/exec"

	"github.com/SolsticeSauer/solsclaw_beta/internal/installer"
)

type Verify struct{}

func (Verify) ID() string                              { return "verify" }
func (Verify) Label() string                           { return "Run `openclaw doctor`" }
func (Verify) ShouldRun(_ installer.StepContext) bool  { return true }

func (s Verify) Run(ctx context.Context, sc installer.StepContext) error {
	cmd := exec.CommandContext(ctx, "openclaw", "doctor")
	out, err := cmd.CombinedOutput()
	if err != nil {
		// `doctor` exits non-zero on warnings too. Surface the report but
		// don't fail the whole pipeline — the user can read the output and
		// decide how to act.
		sc.Bus.Log(s.ID(), "warn", "doctor reported issues:\n"+string(out))
		return nil
	}
	if len(out) == 0 {
		sc.Bus.Log(s.ID(), "info", "doctor reported no issues.")
	} else {
		sc.Bus.Log(s.ID(), "info", string(out))
	}
	return nil
}
